core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -1100,9 +1100,11 @@ class HttpGoogleServicesDAO(val clientSecrets: GoogleClientSecrets,
         billing <- Option(bucketDetails.getBilling)
         rp <- Option(billing.getRequesterPays)
```

Comment:
This makes sense to me as-is. I would be more inclined to store the bucket's location in some Rawls table and eliminate the call to Google entirely … but this would be a roadmap item outside of this PR.

_Maybe_ since we already have the requester-pays setting from workspace settings, and we also have the live requester-pays value from Google, we could use this opportunity to verify them against each other and make sure our workspace settings reflect reality. But we'd only need this if we have worries about data integrity.

Reply:
I'd prefer to pass the user's credentials on this call to Google so we can let Google decide whether they have the necessary access to use the Google project for requester pays. This will guarantee that Terra stays in sync with Google and does not allow users to bill Google projects that they should not be able to. It has the added benefit of simplifying the `getBucketOptions` logic -- we no longer have to connect the Google project to a workspace and check the user's permissions on the workspace, we can just pass the requested Google project along and surface the Google error if needed.

Reply:
After talking about this with Marc and Doug, we decided to keep it as-is.

There are a few wrinkles to using user accounts/pet accounts for credentials, but the main one is that only writers with compute would be able to utilize requester pays workspaces. This would probably make for a confusing user experience; even writers without compute are normally able to incur other types of spend, and it would be weird to cut them off from doing so on requester pays buckets.